.azure-pipelines-steps.yml

```diff
@@ -0,0 +1,50 @@
+# Build steps common to all platforms
+
+steps:
+- task: NodeTool@0
+  inputs:
+    versionSpec: $(node_version)
+  displayName: 'Install Node.js'
+
+- task: Npm@1
+  inputs:
+    command: custom
+    customCommand: i -g npm
+  displayName: 'Update npm'
+  condition: eq(variables['node_version'], '6.x')
+
+- task: Npm@1
+  inputs:
+    command: custom
+    customCommand: ci
+  displayName: 'Install dependencies'
+
+- bash: |
+    npm test -- -- --reporter mocha-junit-reporter --reporter-options mochaFile=./test-results.xml
```

Comment:
Yah ... I haven't seen the double "--" before. For future reference, what requires this?

Reply:
It's because we're calling a npm script that calls another npm script. Our params need to pass through both calls. I see its fragility, but I still prefer defining the xml file in the same place where it's consumed (the yml file). Let me know if you think otherwise, alternative would be to duplicate more on the package.json (w/wo ci reporter).